### ISSUE TITLE
[HOLD for payment 2023-04-10] [$1000] Chat - Backtick text should show strikethrough when offline

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#15571]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

- Open the App on an iOS or [Browserstack](https://app-live.browserstack.com/#os=android&os_version=13.0&device=Google+Pixel+7&zoom_to_fit=true&full_screen=true&speed=1)
- Login with any account
- Open any chat
- Send a message using backticks (ex. ````testing````) 
- Disable the internet connection to see the "You appear to be offline" message
- Delete the sent message
- The deleted message should show a strikethrough: it doesn't

https://user-images.githubusercontent.com/51066321/224864733-2b60c50e-a9b0-4177-b2ff-d547cdbf36d0.mp4

Testing on Android Native:

- Open the App on an Android or [Browserstack](https://app-live.browserstack.com/#os=android&os_version=13.0&device=Google+Pixel+7&zoom_to_fit=true&full_screen=true&speed=1)
- Login with any account
- Open any chat
- Send a message using backticks (ex. ````testing````) 
- Disable the internet connection to see the "You appear to be offline" message
- Delete the sent message
- The deleted message should show a strikethrough: it doesn't

<img width="293" alt="image" src="https://user-images.githubusercontent.com/51066321/224869549-291af4e3-0b93-4395-b9e6-21272e8fe3ef.png">

## Expected Result:
The message should show a strikethrough

## Actual Result:
Block of text deleted offline is not crossed out

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [X] iOS / native
- [X] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** iOS 1.2.83

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01a5ee65db931161a1</li>
        <li>Upwork Job ID: 1635455514550730752</li>
        <li>Last Price Increase: 2023-03-14</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @Christinadobrzyn (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by Christinadobrzyn** [comment]
I think this is a duplicate of - https://github.com/Expensify/App/issues/13986

@jasperhuangg would you be willing to confirm?

**Comment by jasperhuangg** [comment]
Hmm not a duplicate. @Christinadobrzyn 

When a message is deleted offline we apply ~strikethrough~ styles to the text in the message. It seems like the code font we're using doesn't support these ~strikethrough~ styles, hence it doesn't get crossed out.

**Comment by Christinadobrzyn** [comment]
Ah thank you @jasperhuangg! 

Hey @kbecciv testing this on iOS: 
- I get strikethrough. 

Troubleshooting steps are: 

- Open the App on an iOS or [Browserstack](https://app-live.browserstack.com/#os=android&os_version=13.0&device=Google+Pixel+7&zoom_to_fit=true&full_screen=true&speed=1)
- Login with any account
- Open any chat
- Send a block of text message
- Kill off the internet connection by setting airplane mode
- Delete the sent message
- The deleted message shows a strikethrough

Testing on Android Native: 
- The deleted message is disappearing instead of showing the strikethrough

Troubleshooting steps are: 

- Open the App on an Android or [Browserstack](https://app-live.browserstack.com/#os=android&os_version=13.0&device=Google+Pixel+7&zoom_to_fit=true&full_screen=true&speed=1)
- Login with any account
- Open any chat
- Send a block of text message
- Kill off the internet connection - airplane mode
- Delete the sent message
- The deleted message should show a strikethrough instead, it's deleting

Can you confirm you see the same for these two devices?

**Comment by kbecciv** [comment]
@Christinadobrzyn Checking with two of mine devices as well, update you shortly

**Comment by kbecciv** [comment]
Reproduced in IOS app - build 1.2.77.1

https://user-images.githubusercontent.com/93399543/222525052-78f9f20f-6bcc-4d5f-81c8-328570e15515.MP4

Reproduced in Android app, build 1.2.77.1

https://user-images.githubusercontent.com/93399543/222525159-aded1e29-801a-46c6-8db3-77e2baef312d.mp4

**Comment by Christinadobrzyn** [comment]
- So this seems to only be affecting Android - all other devices are showing the strikethrough when offline. 
- I'm assigning Eng to see if this is something we want fix now or later.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @iwiznia (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by iwiznia** [comment]
@Christinadobrzyn @kbecciv are the OP steps correct? Specifically:
> The deleted message should show a strikethrough instead, it's deleting

In the videos I don't see it is getting deleted, just that it does not have strike through

> Send a block of text message

Would be better to clarify what this means, ie: send a message enclosed between backticks

**Comment by Christinadobrzyn** [comment]
Sorry @iwiznia I should have asked more questions about the testing steps of this earlier in the GH. Answering your questions: 

> Would be better to clarify what this means, ie: send a message enclosed between backticks

I understand that 'block of text' means backtick but I also tested this on standard text

> In the videos I don't see it is getting deleted, just that it does not have a strike through

Correct, here's what I see: 

Testing on iOS, 

- The backtick text isn't deleted when offline and isn't showing a strike through.
- Standard text - not backtick text - is showing a strikethrough when deleted offline.

Testing on Android, 

- The backtick text isn't deleted when offline and isn't showing a strike through
- Standard text - not backtick text - is deleted when offline

I'm not sure what part of this needs to be fixed or if we want to do a fix now. If we do want to fix some part of this now, I can update the OP with the specific testing details. @kbecciv please feel free to let me know if I'm missing something.

**Comment by iwiznia** [comment]
I was just trying to understand the issue, since the videos and description did not seem to match.

How it should work is basically the same for all those cases: when you are offline and delete a message, it should show strike through.

Can you update the OP to make it clear what is the current and desired behavior please?

**Comment by MelvinBot** [comment]
@Christinadobrzyn, @kbecciv Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by Christinadobrzyn** [comment]
Updated the OP and adding the external label

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~01a5ee65db931161a1

**Comment by MelvinBot** [comment]
Current assignee @Christinadobrzyn is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @Santhosh-Sellavel (`External`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @youssef-lr (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by bernhardoj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

When we delete a code block text while offline, there is no strikethrough on native platform. Actually, this does not affect code block only, but all markdown.

### What is the root cause of that problem?

We already have an `OfflineWithFeedback` component which will pass a strikethrough style when we delete something while offline (`pendingAction` is delete). This works fine for non markdown style as we are receiving the style here.

https://github.com/Expensify/App/blob/a7b6838c0a502f6fcdae2c0f4bcc57c0ad0a72d5/src/pages/home/report/ReportActionItemFragment.js#L121-L124

But when we add a markdown, we use `RenderHTML` here and we can't pass the style.

https://github.com/Expensify/App/blob/a7b6838c0a502f6fcdae2c0f4bcc57c0ad0a72d5/src/pages/home/report/ReportActionItemFragment.js#L109-L117

### What changes do you think we should make in order to solve the problem?


We should wrap the `htmlContent` with `<del>` tag to make it strikethrough when the item is deleted. I'm thinking 2 way to know whether the item is deleted or not.

1. Check the style whether it contains the strikethrough style from `OfflineWithFeedback`.

or

2. Check if we are offline and if there is a `pendingAction` of delete. The pending action will be passed from here (`props.action.pendingAction`)
https://github.com/Expensify/App/blob/a7b6838c0a502f6fcdae2c0f4bcc57c0ad0a72d5/src/pages/home/report/ReportActionItemMessage.js#L29-L43

**Comment by MelvinBot** [comment]
@youssef-lr @Christinadobrzyn @Santhosh-Sellavel @kbecciv this issue was created 2 weeks ago. Are we close to approving a proposal? If not, what's blocking us from getting this issue assigned? Don't hesitate to create a thread in #expensify-open-source to align faster in real time. Thanks!

**Comment by bernhardoj** [comment]
@Santhosh-Sellavel Sorry, I forget to include that. I have updated my proposal.

**Comment by MelvinBot** [comment]
@youssef-lr @Christinadobrzyn @Santhosh-Sellavel @kbecciv this issue is now 3 weeks old. There is one more week left before this issue <a href='https://stackoverflowteams.com/c/expensify/questions/15233/15234#15234'>breaks WAQ</a> and will need to go internal. What needs to happen to get a PR in review this week? Please create a thread in #expensify-open-source to discuss. Thanks!

**Comment by bernhardoj** [comment]
Waiting for the assignment.

**Comment by youssef-lr** [comment]
Solution looks good to me! cc @Christinadobrzyn

**Comment by Christinadobrzyn** [comment]
Hired in [upwork](https://www.upwork.com/ab/applicants/1635455514550730752/job-details)

@bernhardoj for the fix
@Santhosh-Sellavel as C+

External job posting https://www.upwork.com/jobs/~01a5ee65db931161a1

**Comment by bernhardoj** [comment]
Sorry, I mean the GH issue assignment. I will start working on the PR after I get assigned on this issue.

**Comment by bernhardoj** [comment]
I have opened the PR. Maybe we can use the PR open date for the merge timeline bonus.

**Comment by Santhosh-Sellavel** [comment]
You don't need wait as long you have approval from both C+ & CME. This slows down things.
Technically the date your are assigned or hired seems like valid start date for computing for bonus, incase you started early before either hiring or assigning we can use the PR start date.

**Comment by bernhardoj** [comment]
I will definitely keep that in mind! Thanks!

**Comment by MelvinBot** [comment]
@youssef-lr, @Christinadobrzyn, @Santhosh-Sellavel, @kbecciv Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by Santhosh-Sellavel** [comment]
Waiting for production deployment

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.2.93-4 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/16465


If no regressions arise, payment will be issued on 2023-04-10. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@Santhosh-Sellavel / @youssef-lr] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@Santhosh-Sellavel / @youssef-lr] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@Santhosh-Sellavel / @youssef-lr] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@Christinadobrzyn] Determine if we should create a regression test for this bug.
- [ ] [@Santhosh-Sellavel] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@Christinadobrzyn] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by Christinadobrzyn** [comment]
Looks like this is ready to be paid! 

From what I can tell, the PR wasn't merged within 3 days so not qualifying for a speed bonus, can you confirm @youssef-lr @Santhosh-Sellavel?

**Comment by Santhosh-Sellavel** [comment]
Yes, it does not qualify for a speed bonus

**Comment by Santhosh-Sellavel** [comment]
We wouldn't need a regression test case as basic cases should already covers this

**Comment by Christinadobrzyn** [comment]
Sounds good! I think we can close this GH.

**Comment by MelvinBot** [comment]
@youssef-lr @Christinadobrzyn  Be sure to fill out the [Contact List](https://forms.gle/zhdenCB5dE5gKfnw6)!
